stories/index.stories.js

```diff
@@ -72,21 +86,50 @@ export const S2GetMap = () => {
   };
   perform().then(() => {});
 
-  return img;
+  return wrapperEl;
 };
 
 export const S2GetMapProcessing = () => {
-  if (!process.env.STORYBOOK_AUTH_TOKEN) {
-    return '<div>Please set auth token for Processing API (STORYBOOK_AUTH_TOKEN env var)</div>';
+  if (!process.env.CLIENT_ID || !process.env.CLIENT_SECRET) {
+    return "<div>Please set OAuth Client's id and secret for Processing API (CLIENT_ID, CLIENT_SECRET env vars)</div>";
   }
-  setAuthToken(process.env.STORYBOOK_AUTH_TOKEN);
 
   const img = document.createElement('img');
   img.width = '512';
   img.height = '512';
 
+  const wrapperEl = document.createElement('div');
+  wrapperEl.innerHTML = "<h2>GetMap with Processing for Sentinel-2 L2A</h2>";
+  wrapperEl.insertAdjacentElement("beforeend", img);
+
   // getMap is async:
   const perform = async () => {
+    let authToken;
+    const clientId = process.env.CLIENT_ID;
+    const clientSecret = process.env.CLIENT_SECRET;
+
+    await axios({
+      method: 'post',
+      url: 'https://services.sentinel-hub.com/oauth/token',
+      headers: { 'content-type': 'application/x-www-form-urlencoded' },
+      data: `grant_type=client_credentials&client_id=${clientId}&client_secret=${clientSecret}`,
+    })
+      .then(response => {
+        authToken = response.data.access_token;
+        console.log('Auth token retrieved successfully');
+      })
+      .catch(function (error) {
+        console.log('Error occurred:', {
+          status: error.response.status,
+          statusText: error.response.status,
+          headers: error.response.headers,
+          data: error.response.data,
+        });
+        return;
+      });
```

Comment:
Since this code is duplicated in many places, may I suggest creating a common function that would fetch auth token from the service (using client id and secret)?

As we discussed, it would be even better if this function was in the library itself, though this is a bit bigger change...

Reply:
Thanks, true. I moved the code to a common function. I also did that in `example/node/index.js`.